dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobConfigurationDeserializer.java

```diff
@@ -82,14 +84,22 @@ public JobConfiguration deserialize( JsonParser jsonParser,
         boolean enabled = !root.has( ENABLED ) || root.get( ENABLED ).asBoolean();
 
         boolean continuousExecution = root.has( CONTINUOUS_EXECUTION ) && root.get( CONTINUOUS_EXECUTION ).asBoolean();
-        
+
         boolean leaderOnlyJob = root.has( LEAER_ONLY_JOB ) && root.get( LEAER_ONLY_JOB ).asBoolean();
 
         String cronExpression = root.has( CRON_EXPRESSION ) ? root.get( CRON_EXPRESSION ).textValue() : "";
 
-         JobConfiguration jobConfiguration = new JobConfiguration( root.get( NAME ).textValue(), jobType,
+
+        JobConfiguration jobConfiguration = new JobConfiguration( root.get( NAME ).textValue(), jobType,
             cronExpression, jobParameters, continuousExecution, enabled );
          jobConfiguration.setLeaderOnlyJob( leaderOnlyJob );
+
+        JsonNode idNode = root.get( ID );
+        if ( (idNode != null) && !idNode.isNull() && idNode.isValueNode() )
```

Comment:
You dont need the extra paranthesis inside the if statement here. If you do use them, remember spaces between them and their content.